### ISSUE TITLE
Failed to search vectors in Milvus:Index out of range And Failed to load data to Milvus: 'NoneType' object has no attribute 'insert'

### ISSUE THREAD

**Issue by sxvrqy** [milvus-io/bootcamp#643]
Hello everyone, I have two errors when I use Image Similarity Search with object detection. What are the reasons for this? There is no response in web search, the following is my environment.

`# packages in environment at C:\ProgramData\Anaconda3\envs\milvus_od:
#
# Name                    Version                   Build  Channel
absl-py                   0.13.0                    <pip>
aiofiles                  0.7.0                     <pip>
alabaster                 0.7.12                    <pip>
astor                     0.8.1                     <pip>
astunparse                1.6.3                     <pip>
atomicwrites              1.4.0                     <pip>
attrs                     21.2.0                    <pip>
Babel                     2.9.1                     <pip>
ca-certificates           2021.7.5             haa95532_1    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cachetools                4.2.2                     <pip>
certifi                   2021.5.30        py38haa95532_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
chardet                   3.0.4                     <pip>
charset-normalizer        2.0.4                     <pip>
clang                     5.0                       <pip>
click                     7.1.2                     <pip>
colorama                  0.4.4                     <pip>
decorator                 4.4.2                     <pip>
diskcache                 5.2.1                     <pip>
docutils                  0.17.1                    <pip>
fastapi                   0.65.2                    <pip>
flatbuffers               1.12                      <pip>
gast                      0.3.3                     <pip>
google-auth               1.35.0                    <pip>
google-auth-oauthlib      0.4.6                     <pip>
google-pasta              0.2.0                     <pip>
grpcio                    1.37.1                    <pip>
grpcio-tools              1.37.1                    <pip>
h11                       0.12.0                    <pip>
h5py                      3.1.0                     <pip>
idna                      2.8                       <pip>
imagesize                 1.2.0                     <pip>
Jinja2                    3.0.1                     <pip>
joblib                    1.0.1                     <pip>
keras                     2.6.0                     <pip>
Keras-Preprocessing       1.1.2                     <pip>
Markdown                  3.3.4                     <pip>
MarkupSafe                2.0.1                     <pip>
mmh3                      3.0.0                     <pip>
more-itertools            8.9.0                     <pip>
numpy                     1.19.3                    <pip>
oauthlib                  3.1.1                     <pip>
opencv-python             4.5.2.54                  <pip>
openssl                   1.1.1l               h2bbff1b_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
opt-einsum                3.3.0                     <pip>
packaging                 21.0                      <pip>
paddlepaddle              2.1.0                     <pip>
pandas                    1.2.4                     <pip>
Pillow                    8.3.2                     <pip>
pip                       21.2.2           py38haa95532_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
pluggy                    0.13.1                    <pip>
protobuf                  3.17.3                    <pip>
py                        1.10.0                    <pip>
pyasn1                    0.4.8                     <pip>
pyasn1-modules            0.2.8                     <pip>
pydantic                  1.8.2                     <pip>
Pygments                  2.10.0                    <pip>
pymilvus                  2.0.0rc4                  <pip>
pymilvus-orm              2.0.0rc4                  <pip>
PyMySQL                   1.0.2                     <pip>
pyparsing                 2.4.7                     <pip>
pytest                    5.3.4                     <pip>
python                    3.8.3                he1778fa_2    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
python-dateutil           2.8.2                     <pip>
python-multipart          0.0.5                     <pip>
pytz                      2021.1                    <pip>
PyYAML                    5.4.1                     <pip>
requests                  2.22.0                    <pip>
requests-oauthlib         1.3.0                     <pip>
rsa                       4.7.2                     <pip>
scikit-learn              0.24.2                    <pip>
scipy                     1.7.1                     <pip>
setuptools                52.0.0           py38haa95532_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
six                       1.15.0                    <pip>
snowballstemmer           2.1.0                     <pip>
Sphinx                    2.3.1                     <pip>
sphinxcontrib-applehelp   1.0.1                     <pip>
sphinxcontrib-devhelp     1.0.1                     <pip>
sphinxcontrib-htmlhelp    1.0.2                     <pip>
sphinxcontrib-jsmath      1.0.1                     <pip>
sphinxcontrib-qthelp      1.0.2                     <pip>
sphinxcontrib-serializinghtml 1.1.3                     <pip>
sqlite                    3.36.0               h2bbff1b_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
starlette                 0.14.2                    <pip>
tensorboard               2.6.0                     <pip>
tensorboard-data-server   0.6.1                     <pip>
tensorboard-plugin-wit    1.8.0                     <pip>
tensorflow                2.6.0                     <pip>
tensorflow-estimator      2.6.0                     <pip>
termcolor                 1.1.0                     <pip>
threadpoolctl             2.2.0                     <pip>
typing-extensions         3.7.4.3                   <pip>
ujson                     4.1.0                     <pip>
urllib3                   1.25.11                   <pip>
uvicorn                   0.13.4                    <pip>
vc                        14.2                 h21ff451_1    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
vs2015_runtime            14.27.29016          h5e58377_2    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
wcwidth                   0.2.5                     <pip>
Werkzeug                  2.0.1                     <pip>
wheel                     0.37.0             pyhd3eb1b0_1    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
wincertstore              0.2                      py38_0    http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
wrapt                     1.12.1                    <pip>
`

Error message：

2021-09-09 23:04:04,120 | ERROR | milvus_helpers.py | search_vectors | 107 | Failed to search vectors in Milvus:Index out of range
ERROR 0m: Exception in ASGI application

2021-09-10 07:48:36,401 ｜ ERROR ｜ milvus_helpers.py ｜ insert ｜ 66 ｜ Failed to load data to Milvus: 'NoneType' object has no attribute 'insert'
[31mERROR[0m:    Exception in ASGI application
Traceback (most recent call last):
  File "C:\RIS\reverse_image_search\object_detection\server\src\milvus_helpers.py", line 59, in insert
    mr = self.collection.insert(data)
AttributeError: 'NoneType' object has no attribute 'insert'

**Comment by shiyu22** [comment]
The Error:
```
2021-09-09 23:04:04,120 | ERROR | milvus_helpers.py | search_vectors | 107 | Failed to search vectors in Milvus:Index out of range
ERROR 0m: Exception in ASGI application
```
It means that your search is incorrect, you can check the search vectors.

```
2021-09-10 07:48:36,401 ｜ ERROR ｜ milvus_helpers.py ｜ insert ｜ 66 ｜ Failed to load data to Milvus: 'NoneType' object has no attribute 'insert'
�[31mERROR�[0m: Exception in ASGI application
Traceback (most recent call last):
File "C:\RIS\reverse_image_search\object_detection\server\src\milvus_helpers.py", line 59, in insert
mr = self.collection.insert(data)
AttributeError: 'NoneType' object has no attribute 'insert'
```
I think it occurs when Milvus does not connect successfully, so the Milvvus object is None, and it does not have 'insert'. So please check the Milvus connections and you can refer to this issue #636

**Comment by akimisr** [comment]
I have the same problem with you. Maybe you can solve this problem with add a code line in milvus_helpers.py.
`self.set_collection(collection_name)`
<img width="523" alt="milvus" src="https://user-images.githubusercontent.com/66706285/133020020-7b81d3a4-e1eb-46f4-a394-9592ac80211d.png">

**Comment by shiyu22** [comment]
@akimisr I think you are right, the code `self.set_collection(collection_name)` is missed, you can refer to [milvus_helper](https://github.com/milvus-io/bootcamp/blob/master/solutions/video_similarity_search/quick_deploy/server/src/milvus_helpers.py#L45) in video_search, and can you submit a pull request to reverse_image_search to become a contributor :)

**Comment by shiyu22** [comment]
Thanks @avats-dev :) Closed by #808
